man/sd_bus_message_open_container.xml

```diff
@@ -0,0 +1,142 @@
+    container type and its contents (as a D-Bus type string), respectively. See
+    <citerefentry><refentrytitle>sd_bus_message_append</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    for more information on valid container types and D-Bus type strings.</para>
+
+    <para><function>sd_bus_message_close_container()</function> closes the container opened with
+    <function>sd_bus_message_open_container()</function>. On success, the write pointer is positioned after
+    the closed container.</para>
+
+    <para><function>sd_bus_message_enter_container()</function> enters the next container of the message
+    <parameter>m</parameter>. Entering a container allows reading its contents with
+    <citerefentry><refentrytitle>sd_bus_message_read</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    and similar functions. <parameter>type</parameter> and <parameter>contents</parameter> represent the
+    container type and its contents (as a D-Bus type string), respectively. See
+    <citerefentry><refentrytitle>sd_bus_message_append</refentrytitle><manvolnum>3</manvolnum></citerefentry>
+    for more information on valid container types and D-Bus type strings.</para>
+
+    <para><function>sd_bus_message_exit_container()</function> exits the scope of the last container entered
+    with <function>sd_bus_message_enter_container()</function>. It can only be called once a container has
+    been fully processed. On success, the read pointer is positioned at the next element in the enclosing
+    container of the container that <function>sd_bus_message_exit_container()</function> left.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para>On success, these functions return a non-negative integer. On failure, they return a negative
+    errno-style error code.</para>
+
+    <refsect2>
+      <title>Errors</title>
+
+      <para>Returned errors may indicate the following problems:</para>
+
+      <variablelist>
+        <varlistentry>
+          <term><constant>-EINVAL</constant></term>
+
+          <listitem><para><parameter>m</parameter> or <parameter>contents</parameter> are
+          <constant>NULL</constant> or <parameter>type</parameter> is invalid.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-EPERM</constant></term>
+
+          <listitem><para>The message <parameter>m</parameter> is already sealed.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-ESTALE</constant></term>
+
+          <listitem><para>The message <parameter>m</parameter> is in an invalid state.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-ENOMEM</constant></term>
+
+          <listitem><para>Memory allocation failed.</para></listitem>
+        </varlistentry>
+      </variablelist>
+    </refsect2>
+  </refsect1>
+
+  <xi:include href="libsystemd-pkgconfig.xml" />
```

Comment:
It'd be nice to add an example here. Something like
```C
int append_strings_to_message(sd_bus_message *m, const char* const* arr) {
    int r;

    r = sd_bus_message_open_container(reply, 'a', "s");
        if (r < 0)
                return r;
      for (const char **s = arr; *s; s++) {
              r = sd_bus_message_append(m, "s", *s);
              if (r < 0)
                  return r;
     }
     return sd_bus_message_close_container(m);
}
```
and a similar one for reading?

See 32c4d2ba627b8f5fccba05ca3558782c5d72a78a for an example how to include an external file... Having it external is nice because it's easy to test compilation. (Also note that examples in man/ are indented with 2 spaces, to fit in narrow terminals more easily.)

Reply:
Added examples.